lib/addons/kubernetes-ingress/index.ts

```diff
@@ -0,0 +1,213 @@
+     * @default true
+     */
+    internetFacing?: boolean;
+
+    /**
+     * Specifies how traffic is routed to pods. Can be either 'ip' or 'instance'.
+     * 'ip' mode is more performant and requires VPC-CNI.
+     * @default 'ip'
+     */
+    targetType?: string;
+
+    /**
+     * Hostname to be used with external DNS services for automatic DNS configuration.
+     */
+    externalDnsHostname?: string;
+
+    /**
+     * Specifies the class of the ingress controller. Used to differentiate between multiple ingress controllers.
+     * @default 'nginx'
+     */
+    ingressClassName?: string;
+
+    /**
+     * Specifies the controller class used for handling ingress in a cluster.
+     */
+    controllerClass?: string;
+
+    /**
+     * Identifier used for leader election during the deployment of multiple ingress controllers.
+     */
+    electionId?: string;
+
+    /**
+     * Determines if the ingress controller should be set as the default controller for handling ingress resources.
+     * @default false
+     */
+    isDefaultClass?: boolean;
+
+    /**
+     * Name of the certificate {@link NamedResourceProvider} to be used for certificate look up. 
+     * @see {@link ImportCertificateProvider} and {@link CreateCertificateProvider} for examples of certificate providers.
+     */
+    certificateResourceName?: string;
+
+    /**
+     * ARN of the AWS Certificate Manager certificate to be used for HTTPS.
+     */
+    certificateResourceARN?: string;
+
+    /**
+     * ARN of the AWS Certificate Manager certificate to be used for HTTPS.
+     * @default "3600"
+     */
+    idleTimeout?: string;
+}
+
+// Set default properties for the add-on
+const defaultProps: KubernetesIngressAddOnProps = {
+    name: "kubernetes-ingress",
+    chart: "ingress-nginx",
+    release: "k8s-ingress",
+    version: "4.10.0",
+    repository: "https://kubernetes.github.io/ingress-nginx",
+    backendProtocol: 'http',
+    crossZoneEnabled: true,
+    internetFacing: true,
+    targetType: 'ip',
+    namespace: 'kube-system',
+    idleTimeout: '3600'
+};
+
+// Define the class for the Kubernetes Ingress Add-On, extending HelmAddOn
+@supportsALL
+export class KubernetesIngressAddOn extends HelmAddOn {
+    private readonly options: KubernetesIngressAddOnProps;
+
+    // Constructor for the class, merging default props with user-defined props
+    constructor(props?: KubernetesIngressAddOnProps) {
+        super({ ...defaultProps, ...props } as HelmAddOnProps);
+        this.options = { ...defaultProps, ...props } as KubernetesIngressAddOnProps;
+    }
+
+    // Dependency decorator to ensure this add-on is deployed after the AWS Load Balancer Controller
+    @dependable(AwsLoadBalancerControllerAddOn.name)
+    async deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const props = this.options;
+
+
+
+        // Setup service annotations based on the properties provided
+        const presetAnnotations: any = {
```

Comment:
Have you tested with network load balancer and these annotations set and external DNS?